ffcx/codegeneration/lnodes.py

```diff
@@ -831,7 +831,15 @@ def as_statement(node):
 
 
 def _math_function(op, *args):
-    return MathFunction(op._ufl_handler_name_, args)
+    name = op._ufl_handler_name_
+    dtype = args[0].dtype
+    if name in ("conj", "real") and dtype == DataType.REAL:
```

Comment:
This seems unrelated?

Reply:
Yes just a merge of Matthew s branch. I will fix.